### PR TITLE
Account for astropy.time API change

### PR DESCRIPTION
**Pull request by sean-lockwood**
Fix for astropy.time.Time out_subfmt='date' no longer being supported in Astropy > 4.0.1 (at least in 4.2.0).
Reported in help call INC0167014.

~Should be tested before merged.~

### CODE REVIEW

**Comment by sean-lockwood** [comment]
Tested locally.  Was able to reproduce the `stis_cti` error with astropy 4.2.0, and was able to fix with this change.

File: refstis/functions.py [hunk 1]
```diff
@@ -254,7 +254,7 @@ def get_start_and_endtimes(input_list):
     times.sort()
     start_mjd = times[0]
     end_mjd = times[-1]
-    times = Time(times, format = 'mjd', scale = 'utc', out_subfmt = 'date').iso
+    times = np.array([x.split()[0] for x in Time(times, format = 'mjd', scale = 'utc').iso])
```
Comment (mmaclay) [inline refstis/functions.py:257]: tested this in a notebook and it works perfectly. With the old line I found the same "date" error just from using just this line in a new environment. Great work finding this so quickly!

**Review by mmaclay — APPROVED** [review]
Awesome job with the quick turnaround here Sean!

File: refstis/functions.py (hunk 1 above)
Comment (sean-lockwood) [reply refstis/functions.py:257]: Thanks, Matt!
I wanted to make sure there weren't any other dependency changes, so I ran the CTI code end-to-end to make sure.  We should implement better test coverage to catch these issues in both packages before it reaches the help desk.

**Pull request merged**